### ISSUE TITLE
Helm 3.3 errors on fabric install

### ISSUE THREAD

**Issue by kaitmore** [DecipherNow/helm-charts#729]
Using Helm 3.3 to install charts is failing on the fabric install with templating errors:

```sh
 helm-charts git:(release-2.3)  helm install fabric fabric --set=global.environment=eks -f global.yaml                             2020-08-20 13:13:11
Error: template: fabric/charts/jwt/templates/jwt.yaml:51:12: executing "fabric/charts/jwt/templates/jwt.yaml" at <include "sidecar.envvars" .>: error calling include: template: fabric/charts/control-api/templates/_envvars.tpl:53:10: executing "sidecar.envvars" at <include "envvar" $args>: error calling include: template: fabric/charts/control-api/templates/_envvars.tpl:36:19: executing "envvar" at <(tpl $e.value $top) ne "">: can't give argument to non-function tpl $e.value $top
```

Downgrading to helm 3.2 resolves the issue

**Comment by Minger** [comment]
https://github.com/helm/helm/issues/8628

**Comment by dontlaugh** [comment]
They closed the issue Ming referenced, and pointed to this one https://github.com/helm/helm/issues/8026

**Comment by dontlaugh** [comment]
```diff
diff --git a/fabric/control-api/templates/_envvars.tpl b/fabric/control-api/templates/_envvars.tpl
index caef9a3..fb10ebc 100644
--- a/fabric/control-api/templates/_envvars.tpl
+++ b/fabric/control-api/templates/_envvars.tpl
@@ -32,12 +30,9 @@
       name: {{ $e.secret }}
       key: {{ $e.key }}
           {{- else if eq $e.type "value" }}
-          {{- /* The following removes any undefined values. At this stage, it means both the local and global values are undefined, so its best to just get rid of it */}}
-            {{- if (tpl $e.value $top) ne "" }}
 - name: {{ $envName }}
   value: {{ tpl $e.value $top | quote }} 
             {{- end }}
-          {{- end }}
 {{- end }}
 
 {{- /*  envvars loops through the global sidecar envvars and generates Kubernetes container env keys for both regular values and secrets from the local sidecar values and from the global values as a backup.
@@ -65,4 +60,4 @@ Most users should use the `indent` or `nindent` functions to automatically inden
       {{- include "envvar" $args }}
     {{- end }}
   {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
```

This patch lets the fabric chart deploy, even with helm 3.3, but I'm unsure of the consequences down the road, since we're calling this in several places.

**Comment by chrisbsmith** [comment]
`release-2.3` branch has been fixed and now supports Helm v3.3.

```console
➜  helm-charts git:(release-2.3) ✗ helm version
version.BuildInfo{Version:"v3.3.4", GitCommit:"a61ce5633af99708171414353ed49547cf05013d", GitTreeState:"clean", GoVersion:"go1.14.9"}
➜  helm-charts git:(release-2.3) ✗ kc get pods
NAME                            READY   STATUS      RESTARTS   AGE
jwt-redis-6b84846ffc-vkdk9      1/1     Running     0          3m49s
mesh-redis-0                    1/1     Running     0          3m49s
control-api-0                   2/2     Running     0          3m49s
jwt-security-69bf59d8d4-fqclc   2/2     Running     2          3m49s
control-85876587f4-x9pj2        1/1     Running     0          3m49s
control-api-init-b7vrh          0/1     Completed   0          3m48s
edge-6c9dbcb4db-glfmx           1/1     Running     0          2m53s
svclb-edge-h7wc2                2/2     Running     0          2m33s
svclb-edge-9wtd5                2/2     Running     0          2m33s
svclb-edge-frbcn                2/2     Running     0          2m33s
svclb-edge-l7h6p                2/2     Running     0          2m33s
svclb-edge-j86vc                2/2     Running     0          2m33s
prometheus-0                    2/2     Running     0          114s
catalog-7f8b58bb8f-nv5b7        2/2     Running     0          114s
postgres-slo-0                  1/1     Running     0          114s
slo-75f6789bf4-xntsc            2/2     Running     0          114s
dashboard-b866b9f65-rmmc6       2/2     Running     0          114s
catalog-init-sspl6              0/1     Completed   0          52s
➜  helm-charts git:(release-2.3) ✗
```
